internal/mode/static/manager.go

```diff
@@ -241,11 +241,8 @@ func StartManager(cfg config.Config) error {
 		updateGatewayClassStatus:      cfg.UpdateGatewayClassStatus,
 	})
 
-	objects, objectLists := prepareFirstEventBatchPreparerArgs(
-		cfg.GatewayClassName,
-		cfg.GatewayNsName,
-		cfg.ExperimentalFeatures,
-	)
+	objects, objectLists := prepareFirstEventBatchPreparerArgs(cfg)
```

Comment:
We need to check if SnippetsFilters are enabled. This boolean and all the other fields we passed to this function live on the cfg. I think it is better to pass the entire cfg, instead of passing another boolean.

Reply:
For my understanding, do you think it would have been better to have passed in the cfg the moment we had added the ExperimentalFeatures boolean? 

I'm hoping to learn how to discern when I should pass in just a couple fields and when I should pass in the whole cfg.

Reply:
> For my understanding, do you think it would have been better to have passed in the cfg the moment we had added the ExperimentalFeatures boolean?

Yes, I do. It depends on the case, but three fields from the same config would warrant just passing the entire config.